suitcase/mongo_normalized/__init__.py

```diff
@@ -90,6 +93,24 @@ def __call__(self, name, doc):
         sanitized_doc = event_model.sanitize_doc(doc)
         return super().__call__(name, sanitized_doc)
 
+    def update(self, name, doc):
+        if name == 'start':
+            schema_validators[DocumentNames.start].validate(doc)
+            current_col = self._run_start_collection
+            revisions_col = self._run_start_collection_revisions
+            old = current_col.find_one({'uid': doc['uid']})
+            old.pop('_id')
+            target_uid_docs = revisions_col.find({'uid': doc['uid']})
+            cur = target_uid_docs.sort([('revision', pymongo.DESCENDING)]).limit(1)
+            try:
+                old['revision'] = next(cur)['revision'] + 1
+            except StopIteration:
+                old['revision'] = 0
```

Comment:
I would like to reserve revision' by checking `doc.keys()` inside `update` function and leave `start` schema unchanged. Something similar like we check argument `name`.

Reply:
That's too late. Suppose that the original document has the key `'revision'` in it, the only first saved from the RunEngine.

Reply:
I assume`md` is validated by schema during execution, right? If yes, I agree to reserve `revision` in schema level.
By keywords `not` of jsonschema?

Reply:
Yes, that would be one way to do it. I think that's too big of a decision to make without more serious thought about all the possible implications. Up to this point we haven't reserved any words in the RunStart document, so this would be a major change.

For now, how about we store revisions like this, with a level of nesting:

```py
{'revision': X,
 'document': {...}
```

where `{...}` is the document that we are currently storing. That way, if `{...}` happens to _also_ have a key named "revision" it won't collide with ours. We can always reorganize this later if we decide to go ahead and reserve the term `revision` in the schema.

Reply:
The top-level document should include a `uid` as well, and it should have the same value as `document['uid']`.

Reply:
You mean something like this? 
```
{'revision': X,
 'document': {...}
 'uid': 'xxx-xxx-xxxx'}
```
Why we want `uid` in top-level?

Reply:
It might make it easier to search on efficiently. Or maybe it’s not necessary and `find({‘document.uid’: uid})` is all we need: